source/extensions/filters/http/adaptive_concurrency/controller/gradient_controller.cc

```diff
@@ -215,10 +215,12 @@ void GradientController::recordLatencySample(MonotonicTime rq_send_time) {
     sample_count = hist_sample_count(latency_sample_hist_.get());
   }
 
-  if (inMinRTTSamplingWindow() && sample_count >= config_.minRTTAggregateRequestCount()) {
+  const auto agg_count = config_.minRTTAggregateRequestCount();
+  if (min_rtt_update_mtx_.TryLock() && inMinRTTSamplingWindow() && sample_count >= agg_count) {
```

Comment:
What happens when inMinRTTSamplingWindow() returns false?

Reply:
Oh wow, that's embarrassing.. Wasn't thinking when I made that edit this morning.